storage/aws/aws.go

```diff
@@ -162,11 +171,32 @@ func New(ctx context.Context, cfg Config, opts ...func(*options.StorageOptions))
 
 				if _, err := r.sequencer.consumeEntries(cctx, DefaultIntegrationSizeLimit, r.integrate, false); err != nil {
 					klog.Errorf("integrate: %v", err)
+					return
+				}
+				select {
+				case r.cpUpdated <- struct{}{}:
+				default:
 				}
 			}()
 		}
 	}()
 
+	go func(ctx context.Context, i time.Duration) {
+		t := time.NewTicker(i)
```

Comment:
This will panic if i is <= 0, should there be a check in `WithCheckpointInterval` to prevent this? I think it's fair not to allow negative values. Should we special case 0 and make it bypass the ticker, such that it always update the checkpoint when a message is sent to `r.cpUpdated`? 

If we do this, we can put default value of `CheckpointInterval` to 0. Right now, it's 10.
Note that the docstring in `log.go` suggests that it's important to set the value, i.e that without taking action it's not set, but a value is actually set regardless in `ResolveStorageOptions`.

Reply:
I'll leave those for a follow-up PR, but I had in mind that each storage impl probably has a minimum interval which it will clamp to if a lower value is passed in (e.g. GCP simply _cannot_ write checkpoints faster than once per second).

WithCheckpointInterval cannot know what those values are so doesn't do any checking.